packages/js/product-editor/src/hooks/use-product-attributes.ts

```diff
@@ -78,27 +77,73 @@ export function useProductAttributes( {
 		};
 	};
 
-	const getAugmentedAttributes = ( atts: ProductAttribute[] ) => {
+	const getAugmentedAttributes = (
+		atts: ProductAttribute[],
+		variation: boolean,
+		startPosition: number
+	) => {
 		return atts.map( ( attribute, index ) => ( {
 			...attribute,
-			variation: isVariationAttributes,
-			position: attributes.length + index,
+			variation,
+			position: startPosition + index,
 		} ) );
 	};
 
 	const handleChange = ( newAttributes: ProductAttribute[] ) => {
-		const augmentedAttributes = getAugmentedAttributes( newAttributes );
-		const otherAttributes = isVariationAttributes
+		let otherAttributes = isVariationAttributes
 			? allAttributes.filter( ( attribute ) => ! attribute.variation )
 			: allAttributes.filter( ( attribute ) => !! attribute.variation );
-		setAttributes( augmentedAttributes );
-		onChange( [ ...otherAttributes, ...augmentedAttributes ] );
+
+		// Remove duplicate global attributes.
+		otherAttributes = otherAttributes.filter( ( attr ) => {
+			if (
+				attr.id > 0 &&
+				newAttributes.findIndex( ( a ) => a.id === attr.id ) !== -1
+			) {
+				return false;
+			}
+			// Local attributes we check by name.
+			if (
+				attr.id === 0 &&
+				newAttributes.findIndex(
+					( a ) => a.name.toLowerCase() === attr.name.toLowerCase()
+				) !== -1
+			) {
+				return false;
+			}
+			return true;
+		} );
```

Comment:
Filter the `otherAttributes` so we don't have duplicates and have the server decide which one to keep and which one to remove.